src/gululog_idx.erl

```diff
@@ -528,6 +528,30 @@ init_cache_from_file(_Version = 1, Tid, SegId, Fd) ->
       init_cache_from_file(1, Tid, SegId, Fd)
   end.
 
+-spec restore_cache_from_latest_entry(filename(), cache()) -> ok.
+restore_cache_from_latest_entry(FileName, Tid) ->
+  SegId = gululog_name:filename_to_segid(FileName),
+  Fd = open_reader_fd(FileName),
+  try file:read(Fd, 1) of
+    eof                 ->
+      ok;
+    {ok, <<Version:8>>} ->
+      ok = restore_cache_from_latest_entry(Version, Tid, Fd, SegId, <<>>)
+  after
+    ok = file:close(Fd)
+  end.
+
+-spec restore_cache_from_latest_entry(logvsn(), cache(), file:fd(), segid(), binary()) -> ok.
+restore_cache_from_latest_entry(_Version = 1, Tid, Fd, SegId, ChunkBin) ->
+  case file:read(Fd, ?FILE_ENTRY_BITS_V1 * ?FILE_READ_CHUNK) of
+    eof ->
+      [ ets:insert(Tid, from_file_entry(1, SegId, <<E:?FILE_ENTRY_BITS_V1>>))
+        || <<E:?FILE_ENTRY_BITS_V1>> <= ChunkBin],
+      ok;
+    {ok, TmpChunkBin} ->
+      restore_cache_from_latest_entry(1, Tid, Fd, SegId, TmpChunkBin)
+  end.
+
```

Comment:
To get the latest entry from index file:
1, read the index file and recording the last time `ChunkBin`;
2, when read return `eof`, inert `ChunkBin` into cache.